sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -959,6 +959,54 @@ class CollationStringExpressionsSuite
     assert(collationMismatch.getErrorClass === "DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE")
   }
 
+  test("Levenshtein expressions with collation") {
+    case class LevenshteinTestCase[R](l: String, r: String, c: String, result: R)
+    Seq(
+      LevenshteinTestCase("", "", "UTF8_BINARY", 0),
+      LevenshteinTestCase("", "something", "UTF8_BINARY", 9),
+      LevenshteinTestCase("a", "a", "UTF8_BINARY", 0),
+      LevenshteinTestCase("a", "A", "UTF8_BINARY", 1),
+      LevenshteinTestCase("a", "a", "UTF8_BINARY_LCASE", 0),
+      LevenshteinTestCase("a", "A", "UTF8_BINARY_LCASE", 0),
+      LevenshteinTestCase("bd", "ABc", "UTF8_BINARY_LCASE", 2),
+      LevenshteinTestCase("Xü", "Ü", "UTF8_BINARY_LCASE", 1),
+      LevenshteinTestCase("Xũ", "Üx", "UTF8_BINARY_LCASE", 2),
+      LevenshteinTestCase("", "something", "UTF8_BINARY_LCASE", 9),
+      LevenshteinTestCase("sOmeThINg", "SOMETHING", "UTF8_BINARY_LCASE", 0),
+      LevenshteinTestCase("sOmeThINg", "SOMETHING", "UNICODE", 5),
+      LevenshteinTestCase("sOmeThINg", "SOMETHING", "UNICODE_CI", 0)
+    ).foreach(c => {
+      withSQLConf(SQLConf.CODEGEN_FACTORY_MODE.key -> "CODEGEN_ONLY") {
+        val query = s"SELECT levenshtein(collate('${c.l}', '${c.c}'), collate('${c.r}', '${c.c}'))"
+        // Result & data type
+        checkAnswer(sql(query), Row(c.result))
+        assert(sql(query).schema.fields.head.dataType.sameType(IntegerType))
+      }
+    })
+  }
+
+  test("Levenshtein expression with collation and threshold") {
+    case class LevenshteinThresholdTestCase(l: String, r: String, c: String, t: Int, result: Int)
+    Seq(
+      LevenshteinThresholdTestCase("", "", "UTF8_BINARY", 0, 0),
+      LevenshteinThresholdTestCase("", "something", "UTF8_BINARY", 0, -1),
+      LevenshteinThresholdTestCase("aaa", "AAA", "UTF8_BINARY_LCASE", 0, 0),
+      LevenshteinThresholdTestCase("a", "b", "UTF8_BINARY_LCASE", 1, 1),
+      LevenshteinThresholdTestCase("Xü", "Ü", "UTF8_BINARY_LCASE", 1, 1),
+      LevenshteinThresholdTestCase("Xũ", "Üx", "UTF8_BINARY_LCASE", 1, -1),
+      LevenshteinThresholdTestCase("sOmeThINg", "SOMETHING", "UNICODE", 0, -1),
+      LevenshteinThresholdTestCase("sOmeThINg", "SOMETHING", "UNICODE", 10, 5),
+      LevenshteinThresholdTestCase("sOmeThINg", "SOMETHING", "UNICODE_CI", 0, 0),
+      LevenshteinThresholdTestCase("sOmeThINg", "SOMETHING", "UNICODE_CI", 10, 0)
+    ).foreach(c => {
+      val query = s"SELECT levenshtein(collate('${c.l}', '${c.c}'), " +
+        s"collate('${c.r}', '${c.c}'), ${c.t})"
+      // Result & data type
+      checkAnswer(sql(query), Row(c.result))
+      assert(sql(query).schema.fields.head.dataType.sameType(IntegerType))
+    })
+  }
+
```

Comment:
We can have more relaxed testing in this file in general, these are fine. However, we should very thoroughly perform unit tests in https://github.com/apache/spark/blob/master/common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationSupportSuite.java for Levenshtein distance. Please look at those examples and add analogous tests.